[docs]: use snippets from `hyperledger/iroha-javascript` (#234)
hyperledger/iroha-2-docs: src/guide/javascript.md

public & private keys (more on that later). Thus, a key-pair generation
could look like this:

<<<@/snippets/js-sdk-2-1-1-key-pair.ts

When you have a key pair, you might create a `Signer` using the key pair:

<<<@/snippets/js-sdk-2-1-2-signer.ts

> Well, now we are able to make signatures with `signer.sign(binary)`!

Reviewer: ```suggestion
Now we're able to make signatures with `signer.sign(binary)`!
```

However, to interact with Iroha, we need to be able to do more than just
signing. We would need to send something to Iroha, like transactions or
queries. `Torii` will help us with that.

> The `Torii` handles HTTP / WebSocket communications with Iroha. We will use

Reviewer: ```suggestion
`Torii` handles HTTP / WebSocket communications with Iroha. We will use
```

it to communicate with Iroha endpoints. With the help of `Torii` we can:

- Submit transactions with `Torii.submit()`
- Send queries with `Torii.request()`
- Listen for events with `Torii.listenForEvents()`
- Listen for blocks stream with `Torii.listenForBlocksStream()`
- and so on

`Torii` is just a stateless object which is a compendium of methods. You

> can look at it as it is a class with only static methods. Each method has

Reviewer: ```suggestion
can look at it as if it were a class with only static methods. Each method has
```

> its own _requirements_ to be passed in &mdash; some of them need only an

Reviewer: ```suggestion
its own _requirements_ to be passed in &mdash; some of them only need an
```

:::

> And here is the sample of full `Torii` requirements in Browser:

Reviewer: ```suggestion
And here is a sample of full `Torii` in-Browser requirements:
```

╰───┴──────────────────────────────╯
```

:::code-group

### config.json

<<<@/snippets/js-sdk-8-config.json

### crypto.ts

<<<@/snippets/js-sdk-8-crypto.ts

### client.ts

<<<@/snippets/js-sdk-8-client.ts

### components/StatusChecker.vue

<<<@/snippets/js-sdk-8-components-StatusChecker.vue

### components/CreateDomain.vue

<<<@/snippets/js-sdk-8-components-CreateDomain.vue

### components/EventListener.vue

<<<@/snippets/js-sdk-8-components-EventListener.vue

### App.vue

<<<@/snippets/js-sdk-8-App.vue

### main.ts

<<<@/snippets/js-sdk-8-main.ts

:::

:::info

> In `client.ts`, we don't import `config.json`, but do this instead:

Reviewer 2: ```suggestion
In `client.ts`, we import the configuration file like this:
```

```ts
import { client_config } from '../../config'
```

It is a detail of how the source code of this application really works. For
simplicity, read this line as the following:

> ```ts
> import client_config from 'config.json'
> ```

Reviewer 2: ```suggestion
```

╰───┴──────────────────────────────╯
```

> :::code-group

Reviewer 2: do you think it should be a single code-group? I'd split it in multiple ones, e.g. we can have a code group for `components/` - it makes sense, thematically. Just putting all the files for the app in a single group doesn't seem like a good choice to me.

Author: Actually, I don't consider the whole 8th section useful at all, because in comparison with the rest of this tutorial it has (and had) too many code with almost no detailed explanation. I am not sure, but maybe it makes sense just to leave a link to the source code so the reader might follow it and explore on their own.

Reviewer 2: I think it's useful for some users to have this info displayed like this, and others can go look at the project sources

### config.json

<<<@/snippets/js-sdk-8-config.json

### crypto.ts

<<<@/snippets/js-sdk-8-crypto.ts

### client.ts

<<<@/snippets/js-sdk-8-client.ts

### components/StatusChecker.vue

<<<@/snippets/js-sdk-8-components-StatusChecker.vue

### components/CreateDomain.vue

<<<@/snippets/js-sdk-8-components-CreateDomain.vue

### components/EventListener.vue

<<<@/snippets/js-sdk-8-components-EventListener.vue

### App.vue

<<<@/snippets/js-sdk-8-App.vue

### main.ts

<<<@/snippets/js-sdk-8-main.ts

:::

:::info

In `client.ts`, we don't import `config.json`, but do this:

```ts
import { client_config } from '../../config'
```

It is a detail of how the source code of this application really works. For

> simplicity, read this line as the following:

Author: I added this line to avoid possible confusion like "why do they import the config from `../../config`? In the project's directory structure there is only `config.json`, and no any mention of `../../config`..."

> It is a detail of how the source code of this application really works. For
> simplicity, read this line as the following:

Reviewer 2: ```suggestion
Note that you need to import the config in this way because this is how the source code of this application works. You can interpret this line as `import client_config from 'config.json'`.
```
